fsthttp/request.go

```diff
@@ -402,12 +401,25 @@ func (req *Request) constructABIRequest() error {
 	return nil
 }
 
-func (req *Request) sendABIRequestAsyncStreaming(backend string) (*fastly.PendingRequest, *fastly.HTTPBody, error) {
+func (req *Request) sendABIRequestAsync(backend string) (*fastly.PendingRequest, *fastly.HTTPBody, error) {
 	if req.sent {
 		return nil, nil, fmt.Errorf("request already sent")
 	}
 
-	abiPending, err := req.abi.req.SendAsyncStreaming(req.abi.body, backend)
+	// When SendAsyncStreaming is called, it streams the request body to
+	// the backend using "Transfer-Encoding: chunked".  For requests
```

Comment:
Note that this is only the case with the default framing headers mode; see https://docs.rs/fastly/latest/fastly/http/enum.FramingHeadersMode.html

This is definitely the right change to make, but for applications which need to write to the body after sending the headers _and_ want a `content-length`, the manual framing headers mode is available.

Reply:
Good point, I'll add this nuance to the comments.  (We support framing headers mode in the SDK.)